Everest03.NET/Services/Service.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Everest03.NET.Services
+{
+    public class Service : IService
+    {
+        private readonly List<Customer> _customers;
+        
+        public Service(List<Customer> customers)
+        {
+            _customers= customers ?? throw new ArgumentNullException(nameof(customers));
+        }
+
+        private long _id = 1;
+
+        public long SetCustomer(Customer customer)
+        {
+            customer.Cpf = new Regex("[.-]").Replace(customer.Cpf, string.Empty);
+            EmailAlreadyExists(customer.Email);
+            CpfAlreadyExists(customer.Cpf);
+            customer.handle(_id);
+            _customers.Add(customer);
+            _id++;
+            return _customers.Last().Id;
+        }
+
+        public List<Customer> GetCustomers()
+        {
+            return _customers;
+        }
+
+        public Customer GetCustomerById(long Id)
+        {
+            IdExists(Id);
+            return _customers.FirstOrDefault(customer => customer.Id == Id)!;
+        }
+        
+        public void DeleteCustomer(long id)
+        {
+            IdExists(id);
+            _customers.RemoveAll(customer => customer.Id == id);
```

Comment:
Existe a possibilidade de haver mais de 1 customer com o mesmo Id?

Se sim, código está com um buraco na segurança
Se não: 
```suggestion
            _customers.Remove(customer => customer.Id == id);
```